### ISSUE TITLE
Extension does not load in Firefox Private Mode

### ISSUE THREAD

**Issue by ple103** [bitwarden/browser#136]
![screenshot from 2017-04-15 10-06-11](https://cloud.githubusercontent.com/assets/935054/25059191/373f60cc-21c5-11e7-90d6-ea0623e96cf0.png)

**Comment by kspearrin** [comment]
This is a known issue that Mozilla seems unwilling to fix. It is due to the popup being unable to communicate with the background page during private mode. Mozilla claims that inter-window communication should not be allowed in private mode (seems to be in Chrome and others?). This is a pretty fundamental architectural decision that the bitwarden app uses (mostly because it is the encouraged way to build complex web extensions).

I don't have a possible for the time being.

**Comment by mrdokenny** [comment]
One way to bypass this is to use the right click context menu to fill in logins.

**Comment by ple103** [comment]
@mrdokenny Auto-fill is really all I need so this is helpful. Thank you.

**Comment by mhammond** [comment]
> This is a known issue that Mozilla seems unwilling to fix.

https://bugzil.la/1329304 for completeness

**Comment by kspearrin** [comment]
Closing since there is nothing that I can do for this.

**Comment by AMDphreak** [comment]
Dashlane, Lastpass, RoboForm, and virtually every other password plugin function in Private windows. Any idea what they use?

**Comment by shieldwed** [comment]
As of yesterday (Firefox 67, https://www.mozilla.org/en-US/firefox/67.0/releasenotes/), it seems to be possible to control extensions in private mode. Can this issue be reevaluated?

**Comment by Crocmagnon** [comment]
Please see https://github.com/bitwarden/browser/issues/938

**Comment by jeidsath** [comment]
All that should be necessary is to turn off auto-fill in private mode.

Opening the popup and searching the vault and copying keys should not require inter-window communication. You might even be able to grab the page location from somewhere.

**Comment by sandsou** [comment]
Enabling privacy.firstparty.isolate under _about:config_ causes Bitwarden prompting me it can't be used in private mode, despite that I'm in a normal Firefox window.

**Comment by Draic** [comment]
I am using vivaldi browser, which is chromium based. Since today the bitwarden addon states that it can't load because of firefox private mode despite the fact that I am not using firefox.

**Comment by kspearrin** [comment]
@Draic Seems to work for me.

```

Vivaldi | 2.9.1705.41 (Stable channel) (64-bit)
Revision | ce637bfd730e6b2e549bf8def38f849e1a26bd3b
OS | Windows 10 OS Version 1903 (Build 18362.476)

```

**Comment by jayashrinarayana** [comment]
I'd just like to repeat what @AMDphreak said "Dashlane, Lastpass, RoboForm, and virtually every other password plugin function in Private windows. Any idea what they use?". Why is Bitwarden the only password manage that has issues with Firefox private mode? Every other password manager and their nan has managed to figure it out. I really hope Bitwarden does soon otherwise I'll have to stop using it. Not that it matters however I do pay for the product so I'm not just complaining without supporting.

**Comment by jh72de** [comment]
Wait, nothing but complaining about Firefox's intended functionality in three years? Not even trying to figure out how others are doing it?

**Comment by liberty1tech** [comment]
> Wait, nothing but complaining about Firefox's intended functionality in three years? Not even trying to figure out how others are doing it?

The lead developer / founder disingenuously replied on the matter here, which is where my email support request was directed. I also am a paying user.

I say he replied disingenuously because he did. The reply doesn't acknowledge that it's an issue, that they're working on it, or that other password managers make it work just fine. He simply pointed to Mozilla documentation, as if that meant anything at all.

**Comment by fuomag9** [comment]
It's embarassing that every competitor can do this except for bitwarden

**Comment by MarcusOtter** [comment]
I've been using another password manager for ~2 years and was ready to switch to bitwarden beacuse of it's open source nature. This issue seems to be getting neglected, which is a pretty big turn off for me personally since I do most of my browsing in private mode on Firefox. I'll stick to my old manager for now, thought I'd let you know that some people (myself included) consider it to be a big enough problem to rather use something else.

Other than that, I'm a fan of what you've got going here. Hoping that this problem is still being considered so I may be able to make the switch some time in the future 😄

**Comment by cscharf** [comment]
Reopening.

Adding note from: https://github.com/bitwarden/browser/issues/1050#issuecomment-751793690

**Comment by devurandom** [comment]
I was able to work-around this by going to Firefox > Hamburger menu > Add-ons > Bitwarden > Details tab > "Run in private Windows" and toggling that switch to "Allow" and then back to "Don't allow". Afterwards Bitwarden's extension button / menu again shows the regular content. This procedure has to be repeated every once in a while.

(I think I already wrote it somewhere, but cannot find the comment anymore. Sorry if this happens to be repeated.)

**Comment by Hinton** [comment]
Wanted to give a quick update on this issue. Chrome recently announced that [Manifest V3 will be available in M88 Beta](https://blog.chromium.org/2020/12/manifest-v3-now-available-on-m88-beta.html), which will require a significant re-architecture of the browser extension. This re-architecture would also add proper support for private mode in Firefox. Unfortunately, while investigating the Manifest V3 specification we discovered that there is currently no support for persistent Memory storage which is required for security reasons. There is an [open issue](https://bugs.chromium.org/p/chromium/issues/detail?id=930235) to add support for this feature though, which we are tracking.

Once Chrome adds support for persistent Memory I hope to be able to pick up this issue again.

**Comment by liberty1tech** [comment]
First BitWarden blamed Mozilla for not having the functionality that LastPass and 1Password had. Now BitWarden is blaming Chrome for not having the functionality. Hey guys, look. 1Password and LastPass make this work, so your excuse can't be to point fingers at Mozilla or Chrome. Stop doing it.

- Sincerely, a paying user who won't pay again.

> Wanted to give a quick update on this issue. Chrome recently announced that [Manifest V3 will be available in M88 Beta](https://blog.chromium.org/2020/12/manifest-v3-now-available-on-m88-beta.html), which will require a significant re-architecture of the browser extension. This re-architecture would also add proper support for private mode in Firefox. Unfortunately, while investigating the Manifest V3 specification we discovered that there is currently no support for persistent Memory storage which is required for security reasons. There is an [open issue](https://bugs.chromium.org/p/chromium/issues/detail?id=930235) to add support for this feature though, which we are tracking.
> 
> Once Chrome adds support for persistent Memory I hope to be able to pick up this issue again.

**Comment by StephenBrown2** [comment]
> Now BitWarden is blaming Chrome for not having the functionality.

If you look at the linked issue: https://bugs.chromium.org/p/chromium/issues/detail?id=930235 You'll see that it's actually LastPass that raised it as a "key feature" "no longer be[ing] allowed". So yes, it's actually Chrome's "fault", and LastPass will no longer be able to work with Manifest V3, either.

BitWarden works great for me, and having to right click in Private Mode is not an inconvenience that will make me stop using it. I am sorry that it is for you.

- Sincerely, a paying user who will continue paying.

**Comment by canufrank** [comment]
> BitWarden works great for me, and having to right click in Private Mode is not an inconvenience that will make me stop using it. I am sorry that it is for you.

right-click doesn't always work either. requires toggling "allow in private' as another user suggested. overall, a deal breaker, and will prevent me from redirecting my money from a manger that works.

**Comment by Crocmagnon** [comment]
> requires toggling "allow in private'

This is not only Bitwarden: every Firefox extension is by default disabled in private mode and requires a special action from the user. Same thing in Chrome.

**Comment by superbarney** [comment]
> This is not only Bitwarden: every Firefox extension is by default disabled in private mode and requires a special action from the user. Same thing in Chrome.

Bitwarden does not work in Private Windows on Firefox 85.0.2 (latest to-date) even _after being allowed to run_ in Private Windows. I also tried toggling this multiple times across restarts with no success. Very surprised with this major limitation. I signed up for a Premium account without knowing this. Don't think I would have if I knew earlier. To add a bit of context to some who may be interested - I'm a paying LastPass user looking to migrate.

**Comment by gorman42** [comment]
Quite frankly, while I wish for this to be solved, as a user who just moved from LastPass, after their last "upgrade" fiasco, there is no way in hell that this need to right click would make me go back to, or stay with, LastPass.

I will wait patiently for the necessary missing pieces to be available. Meanwhile I find the workaround more than acceptable.

**Comment by Crocmagnon** [comment]
I also wish to re-state that the keyboard shortcuts work flawlessly in private mode in Firefox : you can use Ctrl (or Cmd on macOS) + Shift + L to autofill your passwords 🙂

**Comment by dan-the-man-nz** [comment]
former lastpass user here, very happy with changing to bitwarden last week,  but yeah this isue is a productivity killer - please implement whatever workaround lastpass etc have done to make this work in firefox private windows, as have been using lastpass in private mode for years with no issue.

**Comment by GirzzlyAK** [comment]
The only thing I get when I right click is a notification that my Vault is Locked WITH NO WAY TO UNLOCK IT, so... not sure how this is a workaround, but maybe I'm missing something. I just rolled back FF to 85.0, which still works and which was a real pain in the arse from Mozilla trying to be helpful and not letting me overwrite a profile with an "old" version. Have to delete the compatibility.ini file in the profile and reset your profile.ini. 

This issue needs to be resolve, or Bitwarden is becoming a non-option, and I'm a premium member.

**Comment by YoshiRulz** [comment]
> The only thing I get when I right click is a notification that my Vault is Locked WITH NO WAY TO UNLOCK IT, so... not sure how this is a workaround, but maybe I'm missing something.

You can unlock the vault from a non-Private Browsing window. Which raises the question: if the two instances can communicate, why not pass whatever it is that's inaccessible in Private Browsing through from the other instance...

**Comment by GirzzlyAK** [comment]
> You can unlock the vault from a non-Private Browsing window.

That doesn't help if you use Always Private Browsing Mode like I do. You can't open non-private tab in that mode. The option is to _not_ use that mode, which is just insane. This broke for me overnight after a silent FF update from 86.0 to 86.1, so whatever changed broke BW. Not sure why I'm seeing references to a post about Chrome dating back to 2017 on this issue about private windows. Has this been a known issue for 5 years and not been resolved?

**Comment by GirzzlyAK** [comment]
So, I just restarted FF (after disabling auto updates) and was congratulated for using the latest version of FF (8.6.0.1)! WTF! I not only told FF NOT to upgrade after just reinstalling v85, but Bitwarden NOW WORKS in private mode,.. in 86.0.1! Just like it did yesterday, but just like it DIDN'T this morning. Jesus...

**Comment by dan-the-man-nz** [comment]
its good its working you but the rest of us its not in 86.0.1

![image](https://user-images.githubusercontent.com/906806/111863394-31f81480-89c0-11eb-9505-e1f9892a493c.png)

**Comment by W6C** [comment]
Can hardly believe there is an unfixed and critical bug dating back to 2017. 
Maybe Lastpass team has a better support for Mozilla products.

**Comment by gorman42** [comment]
> 
> 
> Can hardly believe there is an unfixed and critical bug dating back to 2017.
> Maybe Lastpass team has a better support for Mozilla products.

They really have not better support. The state of the Android app, at least on my phone, is beyond ridiculous. Fully satisfied with the switch to Bitwarden.

**Comment by phamgiahung1368** [comment]
firefox is build with privacy protection in mind , so they dont think this is a bug and need to be fixed

**Comment by canufrank** [comment]
> firefox is build with privacy protection in mind , so they dont think this is a bug and need to be fixed

how is crippling a password manager keeping privacy protection in mind? also, other password managers work including lastpass in private windows - the only thing it shows is that bitwarden development leaves much to be desired.

**Comment by BBaoVanC** [comment]
The message talking about this is appearing in regular Firefox windows for me now.

> Unfortunately this window is not available in private mode for this browser.

**Comment by Avrution** [comment]
Just came across this issue after upgrading to Waterfox G3 - in private mode the box opens blank with just a spinning circle of dots.

**Comment by chucklepie** [comment]
Right-click is only of use if it can match the entry, there may be times when you want to pick a username/password from your list and BitWarden context doesn't allow this. In other password managers 'all' entries are available, but in BitWarden this is only available after you can get to the plugin and select 'Vault'....

**Comment by kawtharrr** [comment]
> > Now BitWarden is blaming Chrome for not having the functionality.
> 
> If you look at the linked issue: https://bugs.chromium.org/p/chromium/issues/detail?id=930235 You'll see that it's actually LastPass that raised it as a "key feature" "no longer be[ing] allowed". So yes, it's actually Chrome's "fault", and LastPass will no longer be able to work with Manifest V3, either.
> 
> BitWarden works great for me, and having to right click in Private Mode is not an inconvenience that will make me stop using it. I am sorry that it is for you.
> 
> * Sincerely, a paying user who will continue paying.

stfu moron

**Comment by Nihal247** [comment]
More than 4 years passed, this issue isn't fixed yet, this made me leave Firefox and choose Brave, because I don't want the headache to start fresh with a new password manager, as no password manager provides the option to import data from Bitwarden.

**Comment by jacek-jablonski** [comment]
Why Enpass was able to overcome it in Firefox incognito mode?


![Zrzut ekranu 2021-08-20 o 23 23 38](https://user-images.githubusercontent.com/35669512/130294795-cdc36aa2-bd53-49b3-b8fa-3c96d8878381.png)

**Comment by RuthlessRuler** [comment]
Any Updates on this issue? is it going to be fixed?

**Comment by altrisi** [comment]
Don't know if this is feasible, but couldn't bitwarden just launch the popup window (like when clicking the top left button) from a regular context when the icon is clicked in a private window, while this is an issue? While not the most convenient, would at least allow use of the extension.

**Comment by kid-oh** [comment]
It's hard for me to imagine that this was a problem five years ago. I just started using Firefox recently. This problem is very disturbing

**Comment by paul-norman** [comment]
For me this is only a minor inconvenience. However, the way that the BitWarden team has communicated their stance on this topic to their users (to whom it is important) is dire and very disappointing - in fact bad enough to get a lurker like me to crawl out from under my rock to comment. On this issue the BitWarden team have basically lied and passed blame repeatedly, despite evidence that a solution was possible being presented repeatedly _(i.e. comparable competitor apps working in private FF windows)_. It really wouldn't have taken much time to write a clear and understandable _(perhaps even acceptable?)_ comment on the matter at some point during the last 5 years. I'll even provide them with a starting point for the future... 

_"Unfortunately the Mozilla team have made a change to Firefox to increase their users' privacy and this has prevented our current extension from working. Sadly the refactoring process to restore previous functionality would require a fundamental rewrite of our core system and would represent a huge time-cost from our core team. Because the Firefox extension is used by a relatively small portion of our users, and private mode by even fewer, we have made the decision not to refactor our entire codebase only to fix this issue. Instead we would rather focus our limited resources on efforts that will bring great features to the most possible users. We understand that this decision will not be popular with those users whom it affects, but please try to understand why we are taking this stance."_

To me it's quite sad that effective, truthful engagement with the community is not one of the top priorities for any company. It doesn't have to be particularly smooth _(we're mostly developers here after all)_, but an honest attempt to communicate really does go a long way. People understand that business decisions need to be made and some won't be popular, just don't lie about them.

**In good news**, it does now seem that with the impending implementation of Chrome Manifest V3 that they will be forced to act. However, again no timeline or priority has been stated on this matter by their team in this thread. However, from the [Google timeline](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/), it would seem that they have a hard deadline for this of **January 2023**. So, at the very longest, this issue should only persist for one more (painful) year.

**Comment by GrizzlyAK** [comment]
I'm not sure why, but I had this issue following an update to FF a while ago (maybe a year?), but did some research, and somehow got it to work again. I use FF exclusively (Win 10), and in permanent private browsing mode (every tab), and update FF with every new release, and my BW Extension works flawlessly! I only rarely open the Desktop App as I have FF open all the time. I'm curious why people are still having this issue, and what am I doing differently?

**Comment by Ascathon** [comment]
@GrizzlyAK The extension does work, but the extension menu in the toolbar doesn't. Workaround in the context menu requires way too many clicks.
Advise to others: hit ctrl (or command)-shift-L to fill out your password (when only 1 entry).

**Comment by GrizzlyAK** [comment]
> but the extension menu in the toolbar doesn't.

you mean _this_ menu? I don't think I've ever even used it.

![image](https://user-images.githubusercontent.com/4411963/149842018-b10afdf2-50f8-4ec7-8411-908223679787.png)

**Comment by Ascathon** [comment]
Left click that button. (in private mode obviously)
![image](https://user-images.githubusercontent.com/21990236/149842281-cd1a5461-d7cd-4b35-aad9-8175064d2e3d.png)

**Comment by GrizzlyAK** [comment]
I AM in private mode.

![image](https://user-images.githubusercontent.com/4411963/149842409-f18a2595-5334-4a26-8388-6de7e6d09359.png)

**Comment by GrizzlyAK** [comment]
When I first experienced BW not working, I saw the blank box like the OP. I don't remember what I did, but I did google it, and found some settings to change in FF and it has been working since, in PM.

**Comment by MCOfficer** [comment]
> When I first experienced BW not working, I saw the blank box like the OP. I don't remember what I did, but I did google it, and found some settings to change in FF and it has been working since, in PM.

Can you go into `about:config`, check "Show only modified preferences" on the right, and play with those values? Perhaps it was one of them.

**Comment by invertostringhe** [comment]
I've just got this error on Firefox, but in a "normal" Window. Also, I don't have any other Firefox instances running on my machine.
![image](https://user-images.githubusercontent.com/12977049/150616027-8a00543e-db98-4e68-9cb4-a93b431082e3.png)
![image](https://user-images.githubusercontent.com/12977049/150616039-54f01fa6-5020-436c-b6f3-f11e37a4f64b.png)

**Comment by MrWizard0001** [comment]
As we approach year 6 of this issue not being resolved, let's be clear about the comment made at the top of this thread by @kspearrin, Founder and CTO:

> This is a known issue that Mozilla seems unwilling to fix.

It is such a disingenuous comment that it strains Bitwarden's credibility as a company. At best, it is an unethical diversion of blame and denial of responsibility. At worst, it misrepresents the truth _so much_ that it could be considered lying.

@paul-norman, [above](https://github.com/bitwarden/browser/issues/136#issuecomment-1014464250), wrote as much in a longer form.

Since **every other major password manager has managed to make this work**, here's what the Founder and CTO _should_ say, if he cared about being truthful.

"This is a known issue that **Bitwarden is unwilling to fix**.

I await a truthful response and apology.

**Comment by cscharf** [comment]
Hello everyone! Chad here with the Bitwarden team.

This has been a pretty contentious issue for some time. Admittedly we don't exactly go back in Github history and revise comments or delete history. We also don't generally apologize for _how_ we understand a given situation at any given point in time. I do see now, however, after re-reading through the myriad posts and comments on this thread a recurring theme that we could certainly have done a better job keeping everyone updated on what we're working on and where we're focused here.

So first, I take personal responsibility for not updating this thread as we discussed or made progress on various avenues on this very topic. From the outside looking in I could understand that being a rather frustrating situation; especially given our understanding of the issue originally.

For 2022, community engagement and **communication** will be coming to the forefront of our engineering and product teams. We're actively hiring for a [platform and community product manager](https://bitwarden.com/careers/4920556003/) who will be helping us focus on providing more transparency and communication on issues exactly like this one to keep our community informed and engaged.

Secondly, the note in regards to [Manifest v3](https://github.com/bitwarden/browser/issues/136#issuecomment-758118565) is absolutely pertinent to this conversation given that we, like other password managers, will need to support this model, and like other password manager browser extensions, will need to work around [certain issues](https://bugs.chromium.org/p/chromium/issues/detail?id=930235). I bring up Manifest v3 because with the way Bitwarden was originally designed, it requires a **substantial** refactor across the board of our browser extension.

The refactor required to support Manifest v3 is almost the same refactor required to support FF’s private mode, namely rewriting the persistent background page and working around those limitations. Other password managers simply designed themselves differently than Bitwarden in the beginning so they ended up "just working" within this limitation, whereas Bitwarden, in order to function, would have needed a substantial re-write; so you can understand some of the hesitation to do that in order to support a single browser's private browsing mode. Not that it's not important, but weighing the resources, cost and effort to do so, would have proven extremely challenging, especially given the size of Bitwarden's engineering team at the time (namely 1).

Fortunately now that we have [no choice](https://github.com/bitwarden/browser/issues/1050#issuecomment-751793690) but to refactor our browser extension, we happen to have the resources and necessary planning going into place to do so.

In the meantime our team member @eliykat has managed to put together a [potential stop-gap](https://github.com/bitwarden/browser/pull/2294) to get things "mostly" working for FF private mode users which we're hoping to push out in March (it's early and rough, more PR details forthcoming). The larger refactor is in the planning stages and we will absolutely need to deliver that this year.

**Comment by gorman42** [comment]
Thanks for your detailed answer. I was among the ones that could not find significant impairment in daily use of Bitwarden while using FF in private mode. But I recognize that everybody has different use cases, so it's good to hear that solutions are being worked on.

On the other hand, reading the "potential stop-gap" you linked to, I noticed this "right-click context menu".
Currently, on version 1.55.0, I can right click and access Bitwarden functionality in FF private mode (autofill, copy password, copy username, password generator). So I don't understand... as it doesn't seem likely that you're working to reduce the functionality of the plugin as a stop-gap measure.

**Comment by GrizzlyAK** [comment]
> Thanks for your detailed answer. I was among the ones that could not find significant impairment in daily use of Bitwarden while using FF in private mode. But I recognize that everybody has different use cases, so it's good to hear that solutions are being worked on.
> 
> On the other hand, reading the "potential stop-gap" you linked to, I noticed this "right-click context menu". Currently, on version 1.55.0, I can right click and access Bitwarden functionality in FF private mode (autofill, copy password, copy username, password generator). So I don't understand... as it doesn't seem likely that you're working to reduce the functionality of the plugin as a stop-gap measure.

Agreed, after initially facing this problem myself following a particular FF update a good while ago, I found here (and perhaps elsewhere) some potential solutions that, once tried, was able to successfully eliminate the problem. BW has been working in FF (Win10) on two different machines in permanent private mode perfectly since that time with NO LOSS of functionality, right-click menu and all.

As suggested following my last comment, I did go through the many about:config settings that have been modified, most being timestamps done by FF itself, but of the ones I tried, I could not reproduce the original problem.  I have few Extensions installed, they are: 

Adblock Plus
Bookmark Manager and Viewer
Bookmark Search Plus 2
Nuke Anything
Web Threat Shield (Webroot AV)
Bitwarden (v1.55.0)

All are set to work in Private Mode.

I do NOT use ANY of FF Logins and Passwords functionality, but do Autofill addresses. Nothing else I have found will trigger the previous bad behavior.

Thanks Chad for the update and good luck to the team on this effort!

**Comment by eliykat** [comment]
@gorman42:
> On the other hand, reading the "potential stop-gap" you linked to, I noticed this "right-click context menu". Currently, on version 1.55.0, I can right click and access Bitwarden functionality in FF private mode (autofill, copy password, copy username, password generator). So I don't understand... as it doesn't seem likely that you're working to reduce the functionality of the plugin as a stop-gap measure.

You're right about that, and I've just confirmed that this will be unaffected by PR #2294. Thanks for pointing that out! The PR is still under development, so some of these may turn out to be non-issues, I just wanted to be careful about what I was promising. I've updated the description to be clearer about this, and we'll have a firmer idea once it goes through QA. If anyone has any further feedback or questions about the PR specifically, I recommend heading over to #2294.

**Comment by francisbarton** [comment]
Just coming here to report the same thing as @invertostringhe - I just for the first time had the Bitwarden Firefox add-on not working in a *normal* (non-Private) tab. I don't have Private tabs open anywhere at the moment.
![image](https://user-images.githubusercontent.com/1819920/152026392-8db3faaf-7905-4c2c-af42-f48395e10e90.png)
This was on https://www.thetrainline.com/
I have add-on version 1.55.0 on Firefox 97.0b7 on Windows 10.

**Comment by eliykat** [comment]
@francisbarton that indicates that the popup cannot contact the background page of the extension - basically it's fallen out of sync. Restarting FF, and/or disabling/enabling the extension, should fix it. If not, you can contact our Customer Success team [here](https://bitwarden.com/contact/) for troubleshooting assistance.

**Comment by MrWizard0001** [comment]
@cscharf, thank you for acknowledging some of BitWarden's failures here, and more importantly, laying out a promise for going forward. I note that [13 months ago](https://github.com/bitwarden/browser/issues/1050#issuecomment-751861583), you made no such acknowledgement, so I'm glad for the change of approach.

I feel like there's still some deflection going on by saying it's because you didn't *go back* and update the thread. Indeed, you were [still blaming Mozilla in Dec 2019](https://github.com/bitwarden/browser/issues/1050#issuecomment-560484218) and again, you closed that issue out just 13 months ago with no acknowledgement.

So it's not about merely failing to communicate. Here's what it is really about, and we as users want you to know it's ok to be honest and say this:

> BitWarden's code is out of date, but we're working on it.

It's that simple. And we'll forgive you for that. You could even add to it, like such:

> BitWarden's code is out of date, but we're working on it. By the way, have we mentioned that we're open source and cheaper than the other password managers?

And that would earn some respect back.

Now, there are some concerns that the reason the browsers are cutting off support for BitWarden's current approach is due to security, so it would be nice to see you suggest a target date sooner than "sometime in 2022." After all, as we've been saying for quite some time, *every other major password manager* has had this working for quite some time.

Overall, yes, thank you for finally acknowledging us. Not so much thank you for trying to spin it as simply being late to communicate. If you had shipped new code, it would have spoken for itself already. We look forward to supporting you and hope that you're able to deliver on your promises soon.

It seems you've been spending quite a bit of money on advertising lately. What could advertise better than a well functioning product that's not unprepared for the future?

Thank you.

**Comment by altrisi** [comment]
I believe repeating this over and over is pointless, because it's not really related to the issue, just empty talking. It's already acknowledged that a fix is in progress and the reasons have been exposed, you can even get the PR (multiple ones actually) and build it if you need it that much (benefits of open source!).

This is an issue tracker, not a forum thread, people here watch issues for updates on them, not to get their email filled with people complaining over and over about how something is an issue, how in the past things (or words) were different or complaining about _my issue is the most important and you're awful for not fixing it as the top priority_.

**Comment by eliykat** [comment]
PR #2294 has been merged, which provides support for FF Private Mode (subject to some limitations listed in the PR). Provided it passes QA, it'll be in the March release.

This the most fully featured workaround we will provide at this time given the significant resources required to rewrite the extension to fully support Private Mode. As [explained above](https://github.com/bitwarden/browser/issues/136#issuecomment-1023663978), unrestricted Private Mode support should come with the manifest v3 rewrite. Until then, these limitations will be noted in our documentation (once updated) and will be considered known/accepted behaviour.

I've closed this thread because this fixes the issue originally reported, i.e. that the extension does not load in FF Private Mode, and there's nothing further we can do here.

Thank you all for bearing with us and I hope you find the limited support useful.
